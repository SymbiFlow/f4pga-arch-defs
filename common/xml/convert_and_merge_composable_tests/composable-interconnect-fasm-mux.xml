<?xml version="1.0"?>
<!-- Test the special handling around the fasm_mux metadata when using
     composable interconnects. -->
<xml>
 <pb_type name="parent">
  <input name="i0" />
  <input name="i1" />
  <output name="o" />

  <pb_type name="child">
   <input name="i" />
   <output name="o" />
  </pb_type>

  <interconnect>
   <direct>
    <port type="output" name="o" />
    <port type="input" from="child" name="o" />
   </direct>
   <mux name="mux1">
    <port type="input" name="i0">
     <metadata>
      <meta name="fasm_mux">a1</meta>
     </metadata>
    </port>
    <port type="input" name="i1">
     <metadata>
      <meta name="fasm_mux">b1</meta>
     </metadata>
    </port>
    <port type="output" from="child" name="i" />
    <metadata>
     <meta name="fasm_name">fasm_name</meta>
    </metadata>
   </mux>
  </interconnect>
 </pb_type>
</xml>
