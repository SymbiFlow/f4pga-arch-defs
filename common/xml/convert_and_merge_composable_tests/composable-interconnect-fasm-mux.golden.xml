<?xml version="1.0"?>
<xml>
  <pb_type name="parent">
    <input name="i0"/>
    <input name="i1"/>
    <output name="o"/>
    <pb_type name="child">
      <input name="i"/>
      <output name="o"/>
    </pb_type>
    <interconnect>
      <direct input="child.o" name="parent-o" output="parent.o"/>
      <mux input="parent.i0 parent.i1" name="mux1" output="child.i">
        <metadata>
          <meta name="fasm_mux">
parent.i0 : a1
parent.i1 : b1
</meta>
          <meta name="fasm_name">fasm_name</meta>
        </metadata>
      </mux>
    </interconnect>
  </pb_type>
</xml>
